pandasai/llm/gpt4all.py

```diff
@@ -0,0 +1,181 @@
+            self._supported_models,
+        )
+        assert self.model_name in self._supported_models, _assert_msg
+
+        # automatically download inside that folder
+        _default_download_path = os.path.join(
+            os.path.expanduser("~"), ".local/share/nomic.ai/GPT4All/"
+        )
+
+        self.model_folder_path = (
+            _default_download_path if model_folder_path is None else model_folder_path
+        )
+
+        # automatically create the default folder and download the model
+
+        if not os.path.exists(self.model_folder_path) and allow_download:
+            os.mkdir(self.model_folder_path)
+
+        if allow_download:
+            self._auto_download()
+            print(f"Model {model_name} saved as: {self.model_folder_path}")
+
+        n_threads = self.n_threads if n_threads is None else n_threads
+        try:
+            from gpt4all import GPT4All
+
+            self.gpt4all_model = GPT4All(
+                model_name=self.model_name,
+                model_path=self.model_folder_path,
+                n_threads=n_threads,
+            )
+        except ImportError:
+            raise ImportError(
+                "Unable to import gpt4all python package "
+                "Please install it with `pip install -U gpt4all`"
+            )
+
+        self.default_parameters = {
+            "max_tokens": self.max_tokens,
+            "n_predict": self.n_predict,
+            "top_k": self.top_k,
+            "top_p": self.top_p,
+            "temp": self.temp,
+            "n_batch": self.n_batch,
+            "repeat_penalty": self.repeat_penalty,
+            "repeat_last_n": self.repeat_last_n,
+        }
+
+        # this will override all the parameters with all the pre-existing ones
+        self.params = {**self.default_parameters, **kwargs}
+
+    def _auto_download(self) -> None:
+        # import all the required dependencies
+        import requests
+        from tqdm import tqdm
+
+        download_path = os.path.join(self.model_folder_path, self.model_name)
+
+        if not os.path.exists(download_path):
+            if self.allow_download:
+                # send a GET request to the URL to download the file.
+                # Stream it while downloading, since the file is large
+
+                try:
+                    url = f"https://gpt4all.io/models/{self.model_name}"
+
+                    response = requests.get(url, stream=True)
+
+                    with open(download_path, "wb") as f:
+                        for chunk in tqdm(
+                            response.iter_content(chunk_size=self.chunk_size)
+                        ):
+                            if chunk:
+                                f.write(chunk)
+
+                except Exception as e:
+                    print(f"=> Download Failed. Error: {e}")
+                    return
+
+                print(f"=> Model: {self.model_name} downloaded sucessfully 🥳")
```

Comment:
```suggestion
                logger.info(
                    f"The model was successfully downloaded in "
                    f"{(int(time.time()) - start_downloading_ts)} seconds."
                )

                print(f"=> Model: {self.model_name} downloaded successfully 🥳")
```
* Log a message about autodownload is finished. 
* Keep the `print()` since it's a user-oriented output.
* Fix typo.